internal/olm/client/client.go

```diff
@@ -88,12 +90,20 @@ type Client struct {
 	KubeClient client.Client
 }
 
-func NewClientForConfig(cfg *rest.Config) (*Client, error) {
-	rm, err := apiutil.NewDynamicRESTMapper(cfg)
+func NewClientForConfig(cfg *rest.Config, httpClient *http.Client) (*Client, error) {
+	rm, err := apiutil.NewDynamicRESTMapper(cfg, httpClient)
```

Comment:
Eh, we already create it previously in the call chain. While I get that it doesn't _need_ to be passed as a parameter I don't think it matters that much and passing it saves us another place where an error can occur deeper in the call stack.

Since this is a nit, going to leave as is

Reply:
Passing pointer down the functions when no data needs to be propagated also brings in the concern that it can be modified accidentally. However, its a nit here, so leaving it to your choice :)